Copy the documentation over to the new location.

medinatiger opened the pull request:
Fixed all the new name and links. The older one will be deleted once we switched everything over.

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
4. If the PR is unfinished, see how to mark it: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

<!-- All reviews default to cc'ing the kube-deploy-reviewers github group. -->
@kubernetes/kube-deploy-reviewers

roberthbailey commented:
The PR description says "move" but I don't see any deletions. Are you copying docs? And if so, do we have a timeline to remove one of the copies that that the two versions don't diverge?

medinatiger commented:
@roberthbailey I mean "copy". I changed the title. It should reduce confusion.

roberthbailey commented on ext-apiserver/gcp-deployer/README.md:
```diff
@@ -0,0 +1,84 @@
+To delete your cluster run `./gcp-deployer delete`
+
+
+### How does the prototype work?
```
Should we update this section to reflect the extension api server in this directory?

medinatiger committed:
Copy the documentation over to the new locaiton.

Fixed the content as necesarry to reflect the new change. The older copy with cluster-api directory will be deleted when everything switch over.

medinatiger commented:
BTW, this PR should be merged after https://github.com/kubernetes/kube-deploy/pull/617 is merged. Otherwise, some links is invalid. Add a hold for that.

/hold

ntfrnzn commented on ext-apiserver/gcp-deployer/CONTRIBUTING.md:
````diff
@@ -0,0 +1,153 @@
+This step is necessary to include the project name (as configured in Google Cloud SDK) in the yaml file.
+
+```bash
+$ ./generate-yaml.sh
````
the generate-yaml.sh script needs to be copied over into this directory, along with machines.yaml.template.

ntfrnzn commented on ext-apiserver/gcp-deployer/CONTRIBUTING.md:
````diff
@@ -0,0 +1,153 @@
+	```bash
+	$ kubectl edit machine $MACHINE_NAME
+	```
+
````
As part of cluster verification, perhaps add comments about inspecting the api that's been deployed, `kubectl get apiservices v1alpha1.cluster.k8s.io -o yaml`, or `curl -k -H "Authorization: bearer $TOKEN" $APISERVER/apis/cluster.k8s.io/v1alpha1/`

jessicaochen commented on ext-apiserver/CONTRIBUTING.md:
```diff
@@ -0,0 +1,57 @@
+*   [Cluster type](https://github.com/kubernetes/kube-deploy/blob/master/ext-apiserver/pkg/apis/cluster/v1alpha1/cluster_types.go#L40)
+*   [Machine type](https://github.com/kubernetes/kube-deploy/blob/master/ext-apiserver/pkg/apis/cluster/v1alpha1/machine_types.go#L42)
+
+### A new Node can be created in a declarative way
```
Should this be Machine? or node machine? (same all through doc)

jessicaochen commented on ext-apiserver/CONTRIBUTING.md:
````diff
@@ -0,0 +1,57 @@
+    upgrade machine
+```
+
+The specific upgrade logic will be implement as part of the machine controller, and is specific to the provider. The user provided provider config will be in `machine.Spec.ProviderConfig`. Users are free to use `machine.ObjectMeta.Annotations` as storage.
````
I do not think we need to call out using ObjectMeta as storage. It is what we are doing short term but I do not want to explicitly call it out as a pattern for long term for all controllers.

medinatiger commented:
/hold remove

medinatiger commented:
/hold cancel

karan commented:
You need the .gitignore file here as well. See https://github.com/kubernetes/kube-deploy/pull/626/files#diff-78f0c28405d3a949d95a99b6edcf6259

jessicaochen commented on ext-apiserver/gcp-deployer/machines.yaml:
```diff
@@ -10,7 +10,7 @@ items:
         "apiVersion": "gceproviderconfig/v1alpha1",
         "kind": "GCEProviderConfig",
-        "project": "k8s-cluster-api",
+        "project": "fmin-gke-dev",
```
Is it intentional to update the checked in copy with your dev project?

medinatiger committed:
Add .gitignore file

medinatiger commented:
@karan , added a .gitignore file

jessicaochen reviewed (approved):
lgtm

jessicaochen commented:
Feel free to assign to me to trigger automerge after all open comments are addressed.

medinatiger commented:
All comments has been addressed

jessicaochen commented:
/lgtm
